File: app/src/main/java/com/lzy/nestescroll/views/BaseViewHolder.java
package com.lzy.nestescroll.views;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lzy.nestescroll.views.customview.BaseCustomViewModel;
import com.lzy.nestescroll.views.customview.ICustomView;

public class BaseViewHolder extends RecyclerView.ViewHolder {
    ICustomView view;

    public BaseViewHolder(ICustomView view) {
        super((View) view);
        this.view = view;
    }

    public void bind(@NonNull BaseCustomViewModel item) {
        view.setData(item);
    }
}

File: app/src/main/java/com/lzy/nestescroll/views/BaseViewHolder.kt
package com.lzy.nestescroll.views

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.lzy.nestescroll.views.customview.BaseCustomViewModel
import com.lzy.nestescroll.views.customview.ICustomView

class BaseViewHolder<S : BaseCustomViewModel>(var view: ICustomView<S>) :
    RecyclerView.ViewHolder((view as View)) {
    fun bind(item: S) {
        view.setData(item)
    }

}